src/components/SearchResults.js

```diff
@@ -0,0 +1,165 @@
+    setResultContent(resultContentFormatted);
+    setPostId(postId);
+    LocalStorageUtils.persistResult(resultContentFormatted, postId);
+    LocalStorageUtils.persistSearchInput(searchInput);
+  };
+
+  useEffect(() => {
+    // Populate the results from local storage if they exist
+    const {
+      content: currentResultContent, postId: currentResultPostId
+    } = LocalStorageUtils.getResultInfo();
+    if (currentResultContent) {
+      setResultContent(currentResultContent);
+    }
+    if (currentResultPostId) {
+      setPostId(currentResultPostId);
+    }
+
+    const input = LocalStorageUtils.getSearchInput();
+    if (input) {
+      setSearchInput(input);
+      refine(input);
+    }
+  }, []);
+
+  const getResultMatches = (proximity, words) => {
+    return (proximity / words) >= 0.75;
+  }
+
+  const getAIResult = async () => {
+    setIsLoading(true);
+    try {
+      // Check if the algolia results are close enough
+      const hits = results.hits;
+      const resultMatches = hits.length > 0 ? getResultMatches(hits[0]._rankingInfo.proximityDistance, hits[0]._rankingInfo.words) : false;
+      if (resultMatches) {
+        populateSearchResult(hits[0].content, hits[0].post_id, searchInput);
+        return;
+      }
+      const result = await moduleAI.search.getSearchResult(query, "helpcenter");
+      populateSearchResult(result["result"], result["post_id"], searchInput);
+    } catch (exception) {
+      console.log(exception);
+      setNoResult(true);
+    } finally {
+      setIsLoading(false);
+    }
+  };
+
+  const debouncedResults = useMemo(() => {
+    return debounce(function (query) {
+      if (query && query.length === 0) {
+        clear();
+      }
+      refine(query);
+    }, 300);
+  }, []);
+
+  // Clear any debounce problems
+  useEffect(() => {
+    debouncedResults.cancel();
+  }, []);
+
+  if (isLoading) {
+    return (
+      <>
+        <p>searching...</p>
+        <Loader />
+      </>
+    );
+  }
+
+  return (
+    <>
+      <div className="search-container">
+        <button
+          onClick={() => {
+            document.getElementById("search-input-box").focus();
+          }}
+        >
+          <SearchIcon />
+        </button>
+        <input
+          type="text"
+          id="search-input-box"
+          style={{
+            flexGrow: 2,
+          }}
+          value={searchInput}
+          maxLength="144"
+          placeholder="Ask me anything..."
+          onChange={(e) => {
+            setSearchInput(e.target.value);
+            populateSearchResult("", undefined, e.target.value);
+            setNoResult(false);
+            debouncedResults(e.target.value);
+          }}
+          onKeyDown={async (e) => {
+            if (e.key === "Enter") {
+              await getAIResult();
+            }
+          }}
+        />
+      </div>
+      <div className="attribute">
+        <p>
+          <span>{searchInput ? searchInput.length : 0}/144</span>
```

Comment:
@amartya-dev  What is this `144` number? It seems like it should be a dynamic number representing the total results found?

Reply:
That is just the number of characters in the input, we limit that to 144 characters